taskmanager/features/bootstrap/FeatureContext.php

```diff
@@ -114,22 +157,49 @@ public function thereIsTaskNamedAssignedToUserNamed($name, User $user)
     }
 
     /**
-     * @Transform :task
+     * @Given there is a task named :taskName with :status status
      */
-    public function convertTaskNameToTask(string $task): Task
+    public function thereIsTaskNamedWithStatus(string $taskName, Status $status)
     {
-        $taskCollection = $this->taskRegistry->getByName($task);
+        $task = new Task($taskName, $status);
 
-        return reset($taskCollection);
+        $this->taskRegistry = new TaskRegistry();
+        $this->taskRegistry->add($task);
     }
 
     /**
-     * @Transform :user
+     * @When user named :user changes task named :task status to :status
      */
-    public function convertUserNameToUser(string $user): User
+    public function userNamedChangesTaskNamedStatusTo(User $user, Task $task, Status $status)
     {
-        $userCollection = $this->userRegistry->getByName($user);
+        if ($task->getStatus()->equals(Status::closed())) {
```

Comment:
below lines contain a business logic, why did you put it here?
`$task->changeStatus($status, $user);`
and the rest of the logic should be moved to `Task` entity